pkg/smb/nodeserver_test.go

```diff
@@ -241,7 +261,8 @@ func TestNodePublishVolume(t *testing.T) {
 				TargetPath:        smbFile,
 				StagingTargetPath: sourceTest,
 				Readonly:          true},
-			expectedErr: status.Errorf(codes.Internal, "Could not mount target \"./smb.go\": mkdir ./smb.go: not a directory"),
+			expectedErrWindows: nil,
```

Comment:
why it's nil on windows?

Reply:
The CSI-Proxy is not returning any error for the above. Here are the CSI-Proxy logs:
```
I0925 20:14:06.058327    3480 main.go:51] Starting CSI-Proxy Server ...
I0925 20:14:06.058327    3480 main.go:52] Version: v0.2.0-0-ga37f58f
I0925 20:14:11.417233    3480 server.go:122] calling PathExists with path "C:\\var\\lib\\kubelet\\smb.go"
I0925 20:14:11.418235    3480 server.go:220] calling IsMountPoint with path "C:\\var\\lib\\kubelet\\smb.go"
I0925 20:14:11.420241    3480 server.go:122] calling PathExists with path "C:\\var\\lib\\kubelet\\smb.go"
I0925 20:14:11.427243    3480 server.go:172] calling Rmdir with path "C:\\var\\lib\\kubelet\\smb.go"
I0925 20:14:11.428252    3480 server.go:193] calling LinkPath with targetPath "C:\\var\\lib\\kubelet\\smb.go" sourcePath "C:\\var\\lib\\kubelet\\source_test"
I0925 20:14:11.429237    3480 server.go:122] calling PathExists with path "C:\\var\\lib\\kubelet\\target_test"
I0925 20:14:11.430255    3480 server.go:122] calling PathExists with path "C:\\var\\lib\\kubelet\\target_test"
I0925 20:14:11.432234    3480 server.go:172] calling Rmdir with path "C:\\var\\lib\\kubelet\\target_test"
I0925 20:14:11.438235    3480 server.go:193] calling LinkPath with targetPath "C:\\var\\lib\\kubelet\\target_test" sourcePath "C:\\var\\lib\\kubelet\\error_mount_source"
I0925 20:14:11.439234    3480 server.go:122] calling PathExists with path "C:\\var\\lib\\kubelet\\target_test"
I0925 20:14:11.439234    3480 server.go:220] calling IsMountPoint with path "C:\\var\\lib\\kubelet\\target_test"
I0925 20:14:11.440235    3480 server.go:122] calling PathExists with path "C:\\var\\lib\\kubelet\\target_test"
I0925 20:14:11.441235    3480 server.go:172] calling Rmdir with path "C:\\var\\lib\\kubelet\\target_test"
I0925 20:14:11.441235    3480 server.go:193] calling LinkPath with targetPath "C:\\var\\lib\\kubelet\\target_test" sourcePath "C:\\var\\lib\\kubelet\\source_test"
I0925 20:14:11.442243    3480 server.go:122] calling PathExists with path "C:\\var\\lib\\kubelet\\false_is_likely_exist_target"
I0925 20:14:11.443232    3480 server.go:220] calling IsMountPoint with path "C:\\var\\lib\\kubelet\\false_is_likely_exist_target"
I0925 20:14:11.443232    3480 server.go:122] calling PathExists with path "C:\\var\\lib\\kubelet\\false_is_likely_exist_target"
I0925 20:14:11.444234    3480 server.go:172] calling Rmdir with path "C:\\var\\lib\\kubelet\\false_is_likely_exist_target"
I0925 20:14:11.444234    3480 server.go:193] calling LinkPath with targetPath "C:\\var\\lib\\kubelet\\false_is_likely_exist_target" sourcePath "C:\\var\\lib\\kubelet\\source_test"
I0925 20:14:11.445234    3480 server.go:122] calling PathExists with path "C:\\var\\lib\\kubelet\\target_test"
I0925 20:14:11.445234    3480 server.go:220] calling IsMountPoint with path "C:\\var\\lib\\kubelet\\target_test"
I0925 20:14:11.446233    3480 server.go:122] calling PathExists with path "C:\\var\\lib\\kubelet\\target_test"
I0925 20:14:11.454239    3480 server.go:172] calling Rmdir with path "C:\\var\\lib\\kubelet\\target_test"
I0925 20:14:11.454239    3480 server.go:193] calling LinkPath with targetPath "C:\\var\\lib\\kubelet\\target_test" sourcePath "C:\\var\\lib\\kubelet\\source_test"
```

Reply:
Let me find out the exact reason for this behavior and will get back to you.

Reply:
@andyzhangx This is because, on windows during `Mount` operation is called, under the hood we are actually calling linkPath. https://github.com/kubernetes-csi/csi-driver-smb/blob/master/pkg/mounter/safe_mounter_windows.go#L96. Linking a filepath to a directory which will succeed. Where as in unix we are actually performing an mount from file to a folder which will fail. 

Do you have any suggestions on how this can fixed?

Reply:
mount behavior between Linux and Windows is a little different, that's ok. Could you write code like this:
```
[]struct {
		desc               string
		req                csi.NodePublishVolumeRequest
		expectedErr   	   error
		expectedErrLinux   error
		expectedErrWindows error
	}
...
expectedErr := test.expectedErr
if test.expectedErrWindows != test.expectedErrLinux {
	if runtime.GOOS == "windows" {
		expectedErr = test.expectedErrWindows
	} else {
		expectedErr = test.expectedErrLinux
	}
}
```

So in most cases, we only need to define `expectedErr`, and only need to define `expectedErrLinux` and `expectedErrWindows` when it's different. That would save test code.